consensus/istanbul/core/core.go

```diff
@@ -462,81 +462,103 @@ func (c *core) getPreprepareWithRoundChangeCertificate(round *big.Int) (*istanbu
 	return request, roundChangeCertificate, nil
 }
 
-// startNewRound starts a new round. if round equals to 0, it means to starts a new sequence
+// startNewRound starts a new round with the desired round
 func (c *core) startNewRound(round *big.Int) error {
-	roundChange := false
+	logger := c.newLogger("func", "startNewRound", "tag", "stateTransition")
+	prevBlock := c.current.Sequence().Uint64() - 1
+	blockAuthor := c.backend.AuthorForBlock(prevBlock)
```

Comment:
I would move these two lines to the same block as `valSet`, since they're only used there.